clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackend.java

```diff
@@ -54,6 +56,7 @@ public class IsolatedIngestionBackend extends DefaultIngestionBackend
   private final MainIngestionRequestClient mainIngestionRequestClient;
   private final MainIngestionMonitorService mainIngestionMonitorService;
   private final VeniceConfigLoader configLoader;
+  private final ExecutorService completionReportHandlingExecutor = Executors.newFixedThreadPool(10);
```

Comment:
Those 10 threads will be per JVM or per store?

Reply:
It will be per JVM as it is the whole backend thing. 

(Otherwise we will be having thread explosion as we have hundreds of store per server :) )